ikarus/finiteelements/mechanics/materials/vanishingstrain.hh

```diff
@@ -0,0 +1,180 @@
+struct VanishingStrain : public Material<VanishingStrain<strainIndexPair, MI>>
+{
+  using Underlying         = MI;                              ///< The underlying material type.
+  using ScalarType         = typename Underlying::ScalarType; ///< Scalar type.
+  using MaterialParameters = typename Underlying::MaterialParameters;
+
+  static constexpr auto fixedPairs        = strainIndexPair;                     ///< Array of fixed stress components.
+  static constexpr auto freeVoigtIndices  = createfreeVoigtIndices(fixedPairs);  ///< Free Voigt indices.
+  static constexpr auto fixedVoigtIndices = createFixedVoigtIndices(fixedPairs); ///< Fixed Voigt indices.
+  static constexpr auto freeStrains       = freeVoigtIndices.size();             ///< Number of free strains.
+
+  static constexpr auto strainTag          = Underlying::strainTag;          ///< Strain tag.
+  static constexpr auto stressTag          = Underlying::stressTag;          ///< Stress tag.
+  static constexpr auto tangentModuliTag   = Underlying::tangentModuliTag;   ///< Tangent moduli tag.
+  static constexpr bool energyAcceptsVoigt = Underlying::energyAcceptsVoigt; ///< Energy accepts Voigt notation.
+  static constexpr bool stressToVoigt      = true;                           ///< Stress to Voigt notation.
+  static constexpr bool stressAcceptsVoigt = true;                           ///< Stress accepts Voigt notation.
+  static constexpr bool moduliToVoigt      = true;                           ///< Moduli to Voigt notation.
+  static constexpr bool moduliAcceptsVoigt = true;                           ///< Moduli accepts Voigt notation.
+  static constexpr double derivativeFactor = Underlying::derivativeFactor;   ///< Derivative factor.
+
+  /**
+   * \brief Constructor for VanishingStrain.
+   * \param mat The underlying material model.
+   */
+  explicit VanishingStrain(MI mat)
+      : matImpl_{mat} {}
+
+  [[nodiscard]] constexpr static std::string nameImpl() noexcept {
+    auto matName = MI::name() + "_VanishingStrain(";
+    for (auto p : fixedPairs)
+      matName += "(" + std::to_string(p.row) + std::to_string(p.col) + ")";
+    matName += ")";
+    return matName;
+  }
+
+  /**
+   * \brief Returns the material parameters stored in the material
+   */
+  MaterialParameters materialParametersImpl() const { return matImpl_.materialParametersImpl(); }
+
+  /**
+   * \brief Computes the stored energy for the PlaneStrain material.
+   * \tparam Derived The derived type of the input matrix.
+   * \param Eraw The strain mesasure
+   * \return ScalarType The stored energy.
+   */
+  template <typename Derived>
+  ScalarType storedEnergyImpl(const Eigen::MatrixBase<Derived>& E) const {
+    const auto Esol = reduceStrain(E);
+    return matImpl_.storedEnergyImpl(Esol);
+  }
+
+  /**
+   * \brief Computes the strains for the PlaneStrain material.
+   * \tparam voigt A boolean indicating whether to return strains in Voigt notation.
```

Comment:
"Strain" typo might be everywhere?

Reply:
Which typo exactly?

Reply:
Ah planestrain -> vanishingstrain
Yeah it used to be implemtation for planestrain only

Reply:
Ah no I ment that this function does not return strains but stresses.